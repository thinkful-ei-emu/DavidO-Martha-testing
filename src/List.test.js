import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import renderer from 'react-test-renderer';

const card1 ={
  title: 'title1',
  content: 'content of card',
  key: 1
}
const card2 ={
  title: 'title2',
  content: 'content of card',
  key: 2
}

describe('<List />', () =>{
  it('renders without crashing', ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<List key="1" cards={[card1, card2]} header="header1" />, div);
    ReactDOM.unmountComponentAtNode(div);

  });
});

it('renders the UI as expected', () => {
  const snapshot = renderer
    .create(<List key="1" cards={[card1, card2]} header="header1"/>)
    .toJSON();
  expect(snapshot).toMatchSnapshot();  
});
