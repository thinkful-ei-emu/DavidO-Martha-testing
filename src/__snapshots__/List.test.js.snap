// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header1
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        title1
      </h3>
      <p>
        content of card
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        title2
      </h3>
      <p>
        content of card
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
